Hyperion.Tests/ISerializableTests.cs

```diff
@@ -7,9 +7,177 @@
+        {
+            public NotSerializableEvent(bool foo) { }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext context)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public sealed class Person : ISerializable
+        {
+            public Person(string firstName, string lastName)
+            {
+                FirstName = firstName;
+                LastName = lastName;
+            }
+
+            public Person(SerializationInfo info, StreamingContext context)
+            {
+                FirstName = (string)info.GetValue("FirstName", typeof(string));
+                LastName = (string)info.GetValue("LastName", typeof(string));
+            }
+
+            public string FirstName { get; }
+
+            public string LastName { get; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext streamingContext)
+            {
+                info.AddValue("FirstName", FirstName);
+                info.AddValue("LastName", LastName);
+            }
+        }
+
+        public sealed class PersonWithoutConstructor : ISerializable
+        {
+            public PersonWithoutConstructor(string firstName)
+            {
+                FirstName = firstName;
+            }
+
+            public string FirstName { get; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext streamingContext)
+            {
+                info.AddValue("FirstName", FirstName);
+            }
+        }
+
+        public sealed class PersonWithIDeserializationCallback : ISerializable, IDeserializationCallback
+        {
+            public PersonWithIDeserializationCallback() {}
+
+            public PersonWithIDeserializationCallback(SerializationInfo info, StreamingContext context)
+            {
+                FirstName = (string)info.GetValue("FirstName", typeof(string));
+            }
+
+            public string FirstName { get; set; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext streamingContext)
+            {
+                info.AddValue("FirstName", FirstName);
+            }
+
+            public void OnDeserialization(object sender)
+            {
+                throw new Exception("OnDeserialization");
+            }
+        }
+        #endregion
+
+        [Fact]
+        public void CanIgnoreFieldsWithNonSerializedAttribute()
+        {
+            var expected = new NonSerializedTest() {
+                Field1 = 235,
+                Field2 = "non serialized text"
+            };
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<NonSerializedTest>();
+            Assert.Equal(expected.Field1, actual.Field1);
+            Assert.Null(actual.Field2);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void SkipNonSerializedAttributeIfNoSerilizedAttribute()
```

Comment:
`NonSerializedAttribute` should not work, if class if not marked with 'SerializableAttribute'